workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -476,5 +444,89 @@ private static IEnumerable<string> FindDirInPackages(string directory)
 
             return packagePaths.Union(cachedPackagePaths).Distinct();
         }
+
+        internal static void GenerateCodegenRunConfigs()
+        {
+            var toolsConfig = GdkToolsConfiguration.GetOrCreateInstance();
+
+            var schemaCompilerPath = GetSchemaCompilerPath();
+            var logfilePath = toolsConfig.DefaultCodegenLogPath;
+
+            var codegenArgs = new List<string>
+            {
+                $"--json-dir=\"{ImprobableJsonDir}\"",
+                $"--schema-compiler-path=\"{schemaCompilerPath}\"",
+                $"--worker-json-dir=\"{WorkerJsonPath}\"",
+                $"--log-file=\"{logfilePath}\"",
+                $"--descriptor-dir=\"{toolsConfig.DescriptorOutputDir}\"",
+                $"--native-output-dir=\"{toolsConfig.FullCodegenOutputPath}\""
+            };
+
+            if (toolsConfig.VerboseLogging)
+            {
+                codegenArgs.Add("--verbose");
+            }
+
+            // Add user defined schema directories, warn if directory does not exist
+            foreach (var schemaDir in toolsConfig.SchemaSourceDirs)
+            {
+                if (!Directory.Exists(schemaDir))
+                {
+                    Debug.LogWarning($"Schema directory defined in GDK tools configuration does not exist: {schemaDir}");
```

Comment:
Currently if the directory doesn't exist we will get a warning then an error. Do we need this one if we check it inside the code generator? 

The latter should always catch any missing directories?

Reply:
one'd expect they'd be different errors: a mismatch with toolsconfig (which may not be an error) VS a mismatch with a previously valid run config (which imo definitely should be an error)

This check catches cases where a toolsconfig defines something but you can't find it at runconfig generation time - which might not actually be an error e.g. if you never actually made use of the schema in that extra directory.

The check in the code generator would catch everything else e.g. dir exists at run-config generation time but not when the code is actually being generated